GNOME polishing from Q.A findings

worldofpeace opened the pull request:
###### Motivation for this change
Some stuff I thought to fixup when rereading https://github.com/NixOS/nixpkgs/issues/99648

Especially the thing about allowing DE modules to register a flatpak GUI that will be installed automatically.
Though, it's of question if there should be an option to disable that?

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

worldofpeace committed:
nixos/gnome3: don't put epiphany in favorite apps

worldofpeace committed:
nixos/gnome3: don't ship gnome-software

This serves no purpose without flatpak https://github.com/NixOS/nixpkgs/issues/99648#issuecomment-706691174

worldofpeace committed:
nixos/flatpak: introduce guiPackages

This adds basically an indirection to systemPackages
to automatically install an interface for flatpak for their respective
environments. e.g if I enable pantheon and flatpak you'll get appcenter,
and on gnome you'll see gnome-software.

https://github.com/NixOS/nixpkgs/issues/99648#issuecomment-706691174

jtojnar commented on nixos/modules/services/desktops/flatpak.nix:
```diff
@@ -15,6 +15,17 @@ in {
       enable = mkEnableOption "flatpak";
+
+      guiPackages = mkOption {
+        type = types.listOf types.package;
```
Perhaps this should be internal. We might want to drop it when we finish the Nix PackageKit support.

worldofpeace replied:
Sounds good to make it internal, but I'm not sure why we should drop it if there's Nix PackageKit support

worldofpeace commented:
It seems https://github.com/NixOS/nixpkgs/pull/101516/commits/bcd186fb143c10e4516a1752530f4f60070c64bb did nothing... maybe we need an internal option for favorite apps

jtojnar commented on nixos/modules/installer/cd-dvd/installation-cd-graphical-gnome.nix:
```diff
@@ -9,7 +9,15 @@ with lib;
+
+    # Add firefox to shell favorites
+    extraGSettingsOverrides = ''
+      [ org.gnome.shell ]
+      favorite-apps=[ 'firefox.desktop' 'org.gnome.Geary.desktop', 'org.gnome.Music.desktop', 'org.gnome.Photos.desktop', 'org.gnome.Nautilus.desktop', 'org.gnome.Software.desktop' ]
```
```suggestion
      [org.gnome.shell]
      favorite-apps=[ 'firefox.desktop'. 'org.gnome.Geary.desktop', 'org.gnome.Music.desktop', 'org.gnome.Photos.desktop', 'org.gnome.Nautilus.desktop', 'org.gnome.Software.desktop' ]
```

worldofpeace committed:
nixos/gnome3: add favoriteAppsOverride option

Rather messy and only needed for the installation cd, so it's
an internal option.

worldofpeace committed:
nixos/gnome3: add gnome-calendar to favorites

worldofpeace committed:
installation-cd-graphical-gnome: add firefox to favorite-apps

worldofpeace commented:
I had to add an option for favorite-apps, but it's internal since it's only for the installation-cd, and it would be better suited to actually have a better dconf module.

worldofpeace commented:
Also noticed that gnome-calendar should be in favorites as well https://github.com/NixOS/nixpkgs/pull/101516/commits/7df6af303e78986fa208c84f349afd928874fe62

jtojnar commented:
It should not really matter that the option was there twice, I think the issue were the syntax error (missing comma).

worldofpeace commented:
> It should not really matter that the option was there twice, I think the issue were the syntax error (missing comma).

I corrected it to no effect. Maybe I should check again, but I believe I tried building vms with clean states each time.
